docs/dev/concepts.rst

```diff
@@ -1,55 +1,37 @@
+Mozilla in developing a better browser for them. To do this, we need a way to
+instruct Firefox to perform certain actions without having to download an
+update or wait for a new release.
 
-SHIELD is split into three main concepts: Actions, Recipes, and the runtime.
-This document describes those concepts in detail.
+The system is made of two main components, a server and client. These
+components are known simply as the Normandy client and Normandy server. The
+server provides static information that the client acts on.
+
+Normandy's execution is built around two main concepts: Actions, Recipes. This
+document describes those concepts in detail.
 
 .. _actions:
 
 Actions
 -------
-An action is JavaScript code that describes a process to perform within a user's
-browser. The code is designed to run in a sandbox that has no privileged access
-to the rest of the browser except through a :doc:`driver object </dev/driver>`
-that has a constrained API for performing sensitive actions, such as showing a
-survey to a user or getting information about the browser itself.
-
-Actions can accept a configuration object that is specified within the Normandy
-control interface. This configuration, which is part of a
+An action is one of the capabilities provided by Normandy. As of Firefox 66,
+actions are implemented entirely within the browser as native code. The
+server selects an action to run by name, such as "preference-experiment" or
+"show-heartbeat".
+
+Actions can accept a configuration object that is specified within the
+Normandy control interface. This configuration, which is part of a
 :ref:`recipe <recipes>`, contains static data required by the action, such as
 translated strings or boolean options for controlling behavior.
 
-For example, a survey action would include code to check whether we have showed
-a survey to the user recently, to determine which variant of a survey to show,
-and the call to the driver to actually display the survey itself. The
-configuration for the survey action would contain the text of the survey prompt
-and the URL to redirect the user to after answering.
-
-Actions are stored in the Normandy code repository and the
-:ref:`runtime <runtime>` fetches the actions from the Normandy service during
-execution. Here's an example of an action that logs a message to the console:
-
-.. code-block:: javascript
-
-   export default class ConsoleLogAction {
-     constructor(normandy, recipe) {
-       this.normandy = normandy; // The driver object
-       this.recipe = recipe; // Contains recipe arguments and other data
-     }
-
-     // Actions should return a Promise when executed
-     async execute() {
-       this.normandy.log(this.recipe.arguments.message, 'info');
-     }
-   }
-
-   // registerAction is in the sandbox's global object
-   registerAction('console-log', ConsoleLogAction);
+For example, a survey action would include code to check whether we have
```

Comment:
This paragraph is quite disjoint with the previous one which is about action configuration. I think you're trying to summarize the whole subsection. In that case, maybe something like: `For example, a "survey" action consists of code within Firefox that [...]. When issued to clients, this action is paired with configuration that [...].`